From b16661f4dab141ef692ed6c893c0cb05566b64ec Mon Sep 17 00:00:00 2001
From: David Kupka <dkupka@redhat.com>
Date: Fri, 12 May 2017 17:17:05 +0200
Subject: [PATCH] otptoken-add-yubikey: When --digits not provided use default
 value

Since Thin client was introduced default values for options are not populated
in client side plugins. When option has default value and is needed in client
plugin it must be handled by explicitly.

https://pagure.io/freeipa/issue/6900

Reviewed-By: Stanislav Laznicka <slaznick@redhat.com>
---
 ipaclient/plugins/otptoken_yubikey.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ipaclient/plugins/otptoken_yubikey.py b/ipaclient/plugins/otptoken_yubikey.py
index 759b7226819f0fa693e475b25262b5d93ac2d39f..9e0e98996ca2d34f34f61c9c07088a3f544ad166 100644
--- a/ipaclient/plugins/otptoken_yubikey.py
+++ b/ipaclient/plugins/otptoken_yubikey.py
@@ -142,7 +142,10 @@ class otptoken_add_yubikey(Command):
 
         # Write the config.
         cfg = yk.init_config()
-        cfg.mode_oath_hotp(key, kwargs['ipatokenotpdigits'])
+        cfg.mode_oath_hotp(key, kwargs.get(
+            'ipatokenotpdigits',
+            self.get_default_of('ipatokenotpdigits')
+        ))
         cfg.extended_flag('SERIAL_API_VISIBLE', True)
         yk.write_config(cfg, slot=kwargs['slot'])
 
-- 
2.9.3

